File: array/0217_contains_duplicate/0217_contains_duplicate.py
# -*- coding: utf-8 -*-


class Solution(object):
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        d = {}
        for n in nums:
            if n in d:
                return True
            d[n] = True
        return False

nums = [1,2,3,1]
print(Solution().containsDuplicate(nums))

File: array/0217_contains_duplicate/0217_contains_duplicate.go
package main

import (
	"fmt"
)

func containsDuplicate(nums []int) bool {
	m := make(map[int]bool, len(nums))
	for _, n := range nums {
		if m[n] {
			return true
		}
		m[n] = true
	}

	return false
}

func main() {
	nums := []int{1, 2, 3, 1}
	fmt.Println(containsDuplicate(nums))
}
